### ISSUE TITLE
FIDO2 Device Tests

### ISSUE THREAD

**Issue by prusnak** [trezor/trezor-firmware#449]
Create Device Tests for U2F/WebAuthn

This will catch bugs such as https://github.com/trezor/trezor-firmware/issues/448

**Comment by prusnak** [comment]
U2F tests merged to CI via https://github.com/trezor/trezor-firmware/commit/650db2a5c01b80cc745d436f5f4d52b9cc52a93a

**Comment by andrewkozlik** [comment]
Pull requests: https://github.com/solokeys/fido2-tests/pull/19, https://github.com/trezor/trezor-firmware/pull/606

**Comment by andrewkozlik** [comment]
The pull request has been merged into solokeys/fido2-tests. We now need to integrate this into the CI. @tsusanka the steps to get the tests running are as follows: 

1. The tests require the `fido2` package. I wasn't sure whether to add that to the Pipfile in trezor-firmware or whether that should go into some separate Pipfile file in the CI.

2. Clone git@github.com:solokeys/fido2-tests.git.
Note: If solokeys make some updates to the tests which are incompatible with Trezor, they might break our CI. So we should check out a fixed version or maybe create a fork of this repo and use that in the CI, so that we have control over when the changes get introduced, or add this as a submodule to trezor-firmware. I am not sure what the standard solution here is.

3. Launch the tests by running `pytest --sim tests/standard/ --vendor trezor` in the `fido2-tests` root directory.

**Comment by prusnak** [comment]
I completed 1) and 2) in bd79827c524b65951c46ca22b2c255abe653fec7

I leave 3) to @tsusanka

**Comment by tsusanka** [comment]
I have added the tests to CI in b530e7c1053f6ddcd9929ec1f71ba25a0d586eaf. However, it seems to me that the test suite is failing when the randomization is turned on. @andrewkozlik didn't you have a similar experience? I've solved it for now by setting it to [some seed](https://github.com/trezor/trezor-firmware/commit/b530e7c1053f6ddcd9929ec1f71ba25a0d586eaf#diff-7fcd8f4a5caeaad75fdeaca6b937a07dR35), which passes, but is it something we care to fix?
